python/tidy/tidy.py

```diff
@@ -376,6 +391,43 @@ def check_ruff_lints():
             )
 
 
+def split_pyrefly_errors(raw_log: bytes | str) -> Iterator[Tuple[str, int, str]]:
+    if isinstance(raw_log, bytes):
+        raw_log = raw_log.decode("utf-8")
+
+    decoded = raw_log.encode("utf-8").decode("unicode_escape")
+
+    blocks = re.split(r"(?=^ERROR )", decoded, flags=re.MULTILINE)
+
+    for block in blocks:
+        block = block.strip()
+        if not block:
+            continue
+
+        message_match = re.search(r"^ERROR (.*)", block, flags=re.MULTILINE)
+        message = message_match.group(1).strip() if message_match else "<unknown>"
+
+        location_match = re.search(r"-->\s+(.*?):(\d+):\d+", block)
+        if location_match:
+            filename = location_match.group(1).strip()
+            line = int(location_match.group(2))
+        else:
+            filename = "<unknown>"
+            line = 0
+
+        yield normalize_path(filename), line, message
+
+
+def check_pyrefly_type_checking() -> Iterator[Tuple[str, int, str]]:
+    print("\r ➤  Running `pyrefly` checks...")
```

Comment:
Instead of parsing the output, I think you can use the `--output-format json` argument and process it as JSON.